src/NHibernate/Driver/ReflectionBasedDriver.cs

```diff
@@ -46,12 +46,12 @@ protected ReflectionBasedDriver(string providerInvariantName, string driverAssem
 
 			if (connectionType == null || commandType == null)
 			{
-#if NETFX || NETSTANDARD2_1_OR_GREATER
+#if NETFX || NETSTANDARD2_1_OR_GREATER || NET5_0_OR_GREATER 
 				if (string.IsNullOrEmpty(providerInvariantName))
 				{
 #endif
 					throw new HibernateException(string.Format(ReflectionTypedProviderExceptionMessageTemplate, driverAssemblyName));
-#if NETFX || NETSTANDARD2_1_OR_GREATER
+#if NETFX || NETSTANDARD2_1_OR_GREATER || NET5_0_OR_GREATER
 				}
 				var factory = DbProviderFactories.GetFactory(providerInvariantName);
```

Comment:
The reason for this condition was that `DbProviderFactories`was not available in netcoreapp2.0.

Since it is available in NetStandard 2.1, it must be there too in Net Core 2.1, and the precise condition should then be:

```suggestion
#if NETFX || NETSTANDARD2_1_OR_GREATER || NETCOREAPP2_1_OR_GREATER
				}
				var factory = DbProviderFactories.GetFactory(providerInvariantName);
```

Or if it is preferred to write conditions only relative to our actual targets:

```suggestion
#if NETFX || NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
				}
				var factory = DbProviderFactories.GetFactory(providerInvariantName);
```

Reply:
> Since it is available in NetStandard 2.1, it must be there too in Net Core 2.1

No, the first version to implement net standard is .NET Core 3.0: https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-1